### PR TITLE
17 layer extent box

### PR DESCRIPTION
**Pull request by Lenninlasd**
## What does this PR do?

Improves PR #9 .

image
### Related Issue

[#17](https://github.com/boundlessgeo/exchange-search/issues/17)

### CODE REVIEW

**Comment by lasdjs** [comment]
Hi @milafrerichs, How can I run the test internally?

File: src/common/addlayers/AddLayersDirective.js [hunk 1]
```diff
@@ -137,11 +137,14 @@
               layerConfig.CRS = ['EPSG:4326'];
               scope.currentLayer = layerConfig;
               var layer = mapService.createLayerWithFullConfig(layerConfig, scope.currentServerId);
+              mapService.getProjection();
```
Comment (milafrerichs) [inline src/common/addlayers/AddLayersDirective.js]: this line is not needed, right?

File: src/common/map/MapService.js [hunk 2]
```diff
@@ -120,6 +120,53 @@
...
+    }
+    return [[
+      ol.proj.transform([coordinates[0], coordinates[1]], oldProjection, projection),
+      ol.proj.transform([coordinates[0], coordinates[3]], oldProjection, projection),
+      ol.proj.transform([coordinates[2], coordinates[3]], oldProjection, projection),
+      ol.proj.transform([coordinates[2], coordinates[1]], oldProjection, projection)
+    ]];
+  }
+
+  function createBboxVector(coordinates, projection) {
```
Comment (milafrerichs) [inline src/common/map/MapService.js]: rename to:
`createGeoJSONLayerFromCoordinatesWithProjection`

File: src/common/map/MapService.js [hunk 3]
```diff
@@ -120,6 +120,53 @@
...
+        color: 'rgba(0, 0, 255, 0.05)'
+      })
+    });
+    var vectorSource = new ol.source.Vector({
+      features: (new ol.format.GeoJSON()).readFeatures(geojsonObject)
+    });
+    return new ol.layer.Vector({
+      source: vectorSource,
+      style: style
+    });
```
Comment (milafrerichs) [inline src/common/map/MapService.js:153]: For now we're on the old version of OL3 so you can just use the following:

```
return new ol.layer.Vector({
   source: new ol.source.GeoJSON({
     object: geojsonObject
   }),
  style: style
});
```

File: src/common/map/MapService.js [hunk 4]
```diff
@@ -120,6 +120,53 @@
     };
   })();
 
+  function bboxTransfom(coordinates, oldProjection, projection) {
```
Comment (milafrerichs) [inline src/common/map/MapService.js]: rename to:
`createBBoxFromCoordinatesFromProjectionIntoProjection`
and rename the params to: `fromProjection` `toProjection`

File: src/common/map/MapService.js [hunk 5]
```diff
@@ -120,6 +120,53 @@
...
+      fill: new ol.style.Fill({
+        color: 'rgba(0, 0, 255, 0.05)'
+      })
+    });
+    var vectorSource = new ol.source.Vector({
+      features: (new ol.format.GeoJSON()).readFeatures(geojsonObject)
+    });
+    return new ol.layer.Vector({
+      source: vectorSource,
+      style: style
```
Comment (milafrerichs) [inline src/common/map/MapService.js]: move the style into the `AddLayersDirective`
set the style with `bboxLayer.setStyle(bboxStyle());` and create a method `bboxStyle`

File: src/common/map/MapService.js [hunk 6]
```diff
@@ -120,6 +120,53 @@
...
+      ol.proj.transform([coordinates[0], coordinates[1]], oldProjection, projection),
+      ol.proj.transform([coordinates[0], coordinates[3]], oldProjection, projection),
+      ol.proj.transform([coordinates[2], coordinates[3]], oldProjection, projection),
+      ol.proj.transform([coordinates[2], coordinates[1]], oldProjection, projection)
+    ]];
+  }
+
+  function createBboxVector(coordinates, projection) {
+    var geojsonObject = {
+      'type': 'FeatureCollection',
```
Comment (milafrerichs) [inline src/common/map/MapService.js]: We don't need a `FeatureCollection` here. A simple `Feature` does the job as well.

**Comment by milafrerichs** [comment]
thanks for the PR. 
But next time please just update the old one. The branch naming ist not that important. But keeping the conversation in one PR is.

**Comment by milafrerichs** [comment]
and for running tests you should

`grunt karmaconfig` and `grunt karma` after you install everything with `npm install`

File: src/common/addlayers/AddLayersDirective.js (hunk 1 above)
Comment (Lenninlasd) [inline src/common/addlayers/AddLayersDirective.js]: Fixed.

File: src/common/map/MapService.js (hunk 2 above)
Comment (Lenninlasd) [inline src/common/map/MapService.js]: Done.

File: src/common/map/MapService.js (hunk 3 above)
Comment (Lenninlasd) [inline src/common/map/MapService.js:153]: Done.

File: src/common/map/MapService.js (hunk 4 above)
Comment (Lenninlasd) [inline src/common/map/MapService.js]: Done.

File: src/common/map/MapService.js (hunk 6 above)
Comment (Lenninlasd) [inline src/common/map/MapService.js]: Done.

File: src/common/map/MapService.js (hunk 5 above)
Comment (Lenninlasd) [inline src/common/map/MapService.js]: Done.

File: src/common/map/MapService.js (hunk 2 above)
Comment (milafrerichs) [inline src/common/map/MapService.js]: I think you mixed something up. I meant changing `createBboxVector` to `createGeoJSONLayerFromCoordinatesWithProjection`

File: src/common/map/MapService.js [hunk 7]
```diff
@@ -120,50 +120,36 @@
...
-      ol.proj.transform([coordinates[2], coordinates[3]], oldProjection, projection),
-      ol.proj.transform([coordinates[2], coordinates[1]], oldProjection, projection)
+      ol.proj.transform([coordinates[0], coordinates[1]], fromProjection, toProjection),
+      ol.proj.transform([coordinates[0], coordinates[3]], fromProjection, toProjection),
+      ol.proj.transform([coordinates[2], coordinates[3]], fromProjection, toProjection),
+      ol.proj.transform([coordinates[2], coordinates[1]], fromProjection, toProjection)
     ]];
   }
 
   function createBboxVector(coordinates, projection) {
```
Comment (milafrerichs) [inline src/common/map/MapService.js]: change method name to `createGeoJSONLayerFromCoordinatesWithProjection`

**Comment by milafrerichs** [comment]
you changed the wrong method name :)

File: src/common/map/MapService.js (hunk 2 above)
Comment (Lenninlasd) [inline src/common/map/MapService.js]: ohh you're right, I fixed things, I hope it's okay :disappointed_relieved:

**Pull request merged**